### ISSUE TITLE
4.1.3 Status messages and native applications

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag2ict#1]
>  4.1.3 Status Messages (Level AA): **In content implemented using markup languages**, status messages can be programmatically determined through role or properties such that they can be presented to the user by assistive technologies without receiving focus.

that first sentence "In content implemented using markup languages" in practice exempts native applications. I'm not sure this was actually intended? Particularly looking at, say, native apps that show toasts/notifications, this means generally that there's no SC that would apply if those notifications are not announced at all?

**Comment by maryjom** [comment]
Moved to WCAG2ICT repository as the new TF will work to address issues tagged as WCAG2ICT from the WCAG repository.

**Comment by detlevhfischer** [comment]
Whatever the intention of its WCAG authors, today 4.1.3 is included in clause 11 (Software, applicable to native apps) in EN 301 549 v.3.2.1:
> EN 301 549 V3.2.1 (2021-03)
Where ICT is non-web software, it shall satisfy WCAG 2.1 Success Criterion 4.1.3 Status Messages.

**Comment by maryjom** [comment]
@detlevhfischer EN 301 549 also applied 4.1.3 to non-web documents. I'm not sure all documentation types have semantic markup for status messages that can be exposed to AT. There are no notes or caveats in the EN to describe those kinds of issues.  This is why the WCAG2ICT work is important to get done.

**Comment by detlevhfischer** [comment]
@mraccess77  wrote:
> EN 301 549 does refer to the SC - but the SC states in content implemented with markup languages - so if the non-web document or software isn't implemented in markup languages then the criteria doesn't apply.

My point is that by referring directoy to 4.1.3 and its scoping to mark-up languages, the EN authors have created an uinnecessary gap in the Status Messages requirements for apps. It is both technically possible and desirableto expose status messages in native apps.
Which leads to the paradox that the Procedure (in Annex C 11.4.1.3 of the EN 301 549 v.3.2.1) is self-defeating;
>1. Check that the software does not fail WCAG 2.1 Success Criterion 4.1.3 Status messages.
Result
Pass: Check 1 is true
Fail: Check 1 is false
Not applicable: Pre-condition 1 or 2 is not met, or the non-web software does not contain content relevant to G 2.1 Success Criterion 4.1.3 Status messages

If the intention (as it should be) is to apply the 4.1.3 requirement to apps there needs to be a correction here, stating that in its application to software (native apps) the constraint on content implemented in mark-up languages needs to be lifted.

As an aside, it leaves ua with a situation that we may apply 4.1.3 selectively to content in webviews (even if we cannot inspect the source code) while exempting it for native content, which is less than ideal, to put it mildly.

**Comment by bruce-usab** [comment]
This is a good catch about 4.1.3.  With 2.0, it is only 4.1.1 parsing with the <q>content implemented using markup languages</q> exception.  With 2.1, there are three more.

**Comment by mraccess77** [comment]
Before 4.1.3 many were using the language in SC 4.1.2 to flag these types of issues.   The SC includes the text "and notification of changes to these items is available to [user agents](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html#dfn-user-agent), including [assistive technologies](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html#dfn-assistive-technology)."

**Comment by Lboniello** [comment]
@maryjom  what is left for me to do here? Thanks!

**Comment by maryjom** [comment]
@Lboniello When you work on 4.1.3 I wanted you to be aware of this issue and discussion which had previously been opened in the WCAG repository. Hoping this will help with our WCAG2ICT notes for this SC.

**Comment by mitchellevan** [comment]
Proposals for WCAG2ICT

### Option 1: in the spirit of SC 4.1.3

<blockquote>
<p>Word substitution: replace "In content implemented using markup languages" with "In content implemented using markup languages or other technology with support for content change notifications"</p><p>Note 1: A web view in a hybrid app is an example of content implemented using markup languages.</p><p>Note 2: Android and iOS native code are examples of other content technologies with support for content change notifications. On these platforms, native application code can notify assistive technologies that a content change has occurred and should be announced to users.</p>
</blockquote>

Advantages: It's good for users. It's not hard for mobile app developers. It's much easier for black-box testing of apps, where the exact boundaries between native code and web code are usually impossible to determine.

Disadvantage: Does the word substitution exceed what WCAG2ICT can do?

### Option 2: lawyering

<blockquote>
<p>Note 1: A web view in a hybrid app is an example of content implemented using markup languages.</p><p>Note 2: Some non-web content technologies, such as native code in Android and iOS applications, can notify assistive technology that a content change should be announced to users. In these technologies it is a best practice to announce status messages.</p>
</blockquote>

**Comment by maryjom** [comment]
See the proposed guidance in Issue #122. Does it sufficiently address the scenarios you brought up @patrickhlauke?

**Comment by maryjom** [comment]
The TF reached [consensus on the guidance for SC 4.1.3](https://www.w3.org/2023/03/30-wcag2ict-minutes.html#r03) on 30 March. The AG WG [reviewed 4.1.3](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-Third-content-review-by-AGWG2/results#xq11) in the 3rd content review and approved the content on 25 July. If you have questions or comments on the [guidance provided on 4.1.3 Status Messages](https://www.w3.org/TR/wcag2ict/#status-messages) in the WCAG2ICT FPWD, please open a new issue.

**Comment by mitchellevan** [comment]
To call attention to this message https://github.com/w3c/wcag2ict/issues/216#issuecomment-1971842036, I've filed the following issue against EN 301 549: [Apply SC 4.1.3 Status Messages to native code in software \#448](https://labs.etsi.org/rep/HF/en301549/-/issues/448)
